### PR TITLE
Splits MeetupServer into its own module

### LINKED ISSUE #26: The meetup service is implemented as one file
There is an opportunity to make a library here. This would also make this both more testable and potentially easier to integration test.

### PR DESCRIPTION
**Pull request by GabeVillalobos**
Addresses issue #26 

Also flattens the server's internal state into a single struct, in
preparation for a public trait for publisher data stores.

### CODE REVIEW

File: meetup/src/main.rs [hunk 1]
```diff
@@ -190,22 +52,25 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
...
+        .add_service(FindMeServer::new(meetup_server))
         .serve(bind_info);
+
     info!("Started server {}", bind_info);
     server.await?;
 
     Ok(())
 }
+
+pub fn socket_validator(v: String) -> Result<(), String> {
```
Comment (andrewjj20) [inline meetup/src/main.rs:80]: oof demoted to the bottom of the file.

File: meetup/src/main.rs [hunk 2]
```diff
@@ -181,7 +43,7 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         )
         .get_matches();
 
-    let scan_interval = matches.value_of("publisher-scan-interval").unwrap();
+    matches.value_of("publisher-scan-interval").unwrap();
```
Comment (andrewjj20) [inline meetup/src/main.rs]: is this actually not used anywhere?

File: meetup/Cargo.toml [hunk 3]
```diff
@@ -4,6 +4,14 @@ version = "0.1.0"
 publish = false
 edition = "2018"
 
+[lib]
```
Comment (andrewjj20) [inline meetup/Cargo.toml:7]: I didn't think this was needed if the source file was lib.rs

**Review by andrewjj20 — COMMENTED** [review]
There is probably one change in there if that value is not used anywhere.

File: meetup/src/main.rs (hunk 2 above)
Comment (GabeVillalobos) [reply meetup/src/main.rs]: It was being used for the `remove_expired_publishers`. I'll add it back in as a struct property for now.

File: meetup/Cargo.toml (hunk 3 above)
Comment (GabeVillalobos) [reply meetup/Cargo.toml:7]: So it looks like the lib definition is needed, but I can remove the [[bin]] section without any noticeable changes

File: meetup/src/main.rs (hunk 2 above)
Comment (andrewjj20) [reply meetup/src/main.rs]: it looks like this line might not be needed any more. I think things are pretty traceable from the option name.

**Review by andrewjj20 — COMMENTED** [review]
Wanted to add, you might want to test this by running the example.

**Comment by GabeVillalobos** [comment]
Yeah I still need to hand test, but otherwise should have all the changes you mentioned

**Comment by andrewjj20** [comment]
> Yeah I still need to hand test, but otherwise should have all the changes you mentioned

https://github.com/andrewjj20/post/pull/35 should remove the need for you to do that.

**Comment by GabeVillalobos** [comment]
Merged #35 and rebased. Tests should be running on this commit now.

**Review by andrewjj20 — APPROVED** [review]

**Pull request merged**